packages/components/src/color-picker/index.js

```diff
@@ -43,30 +43,144 @@ import Alpha from './alpha';
 import Hue from './hue';
 import Inputs from './inputs';
 import Saturation from './saturation';
-import { colorToState, simpleCheckForValidColor } from './utils';
+import { colorToState, simpleCheckForValidColor, isValidHex } from './utils';
+
+const toLowerCase = ( value ) => String( value ).toLowerCase();
+const isValueEmpty = ( data ) => ( data.source === 'hex' && ! data.hex ) ||
+	( data.source === 'hsl' && ( ! data.h || ! data.s || ! data.l ) ) ||
+	( data.source === 'rgb' && (
+		( ! data.r || ! data.g || ! data.b ) &&
+		( ! data.h || ! data.s || ! data.v || ! data.a ) &&
+		( ! data.h || ! data.s || ! data.l || ! data.a )
```

Comment:
These values above can be zero, so this is causing some trouble with using the keyboard to move through the saturation. If you focus into the saturation handle, you can use <kbd>shift</kbd> + <kbd>left</kbd> or <kbd>shift</kbd> + <kbd>right</kbd> (or just left/right) to move the handle, but you can't get it all the way to the edge.

![colors-key](https://user-images.githubusercontent.com/541093/53823127-1fc4ee00-3f3f-11e9-9821-715b050c6856.gif)

I'm pretty sure this is also [the root of this issue,](https://github.com/WordPress/gutenberg/pull/11772#pullrequestreview-209761283) both seemed fixed if I change this to check for [`isFinite`](https://lodash.com/docs/4.17.11#isFinite)